File: java/hamming/src/main/java/Hamming.java
/**
 * Calculates the Hamming difference between two DNA strands.
 */
public class Hamming {
  /**
   * @param a: A strand to compare to @param b.
   * @param b: A strand to compare to @param a.
   * @return: The Hamming difference between @param a and @param b.
   */
  public static int compute(String a, String b) {
    if (a.length() != b.length()) {
      throw new IllegalArgumentException("Strands must be of equal length.");
    }

    int res = 0;

    for (int i = 0; i < a.length(); i++) {
      if (a.charAt(i) != b.charAt(i)) res++;
    }

    return res;
  }
}

File: go/hamming/hamming.go
/*
Package hamming calculates the Hamming difference between two DNA strands.
*/
package hamming

import "errors"

const testVersion = 5

/*
Distances calculates the Hamming difference between two DNA strands.
*/
func Distance(a, b string) (int, error) {
  dist := 0

  if len(a) != len(b) {
    return -1, errors.New("Strands must be of the same length.")
  }
  for i := 0; i < len(a); i++ {
    if a[i] != b[i] {
      dist += 1
    }
  }

  return dist, nil
}
